Game/MyProject/Game/ResourceManager/ResourceManager.cpp

```diff
@@ -0,0 +1,121 @@
+//
+//  ResourceManager.cpp
+//  RacingGame
+//
+//  Created by Tamas Both - (p) on 17/04/2019.
+//
+
+#include "ResourceManager.hpp"
+
+// -------------
+// PUBLIC GETTER
+// -------------
+
+// returns reference to the already existing static object
+ResourceManager& ResourceManager::getResourceManager() {
+    static ResourceManager resource_manager;
+    return resource_manager;
+}
+
+// resets the resource manager
+void ResourceManager::resetResourceManager() {
+    ResourceManager& resource_manager = getResourceManager();
+    
+    resource_manager.textures.clear();
+    resource_manager.models.clear();
+    resource_manager.texture_cache.clear();
+    resource_manager.model_cache.clear();
+    resource_manager.FALLBACK_TEXTURE.reset();
+}
+
+// --------------------------------
+// PUBLIC FUNCTION(S) AND METHOD(S)
+// --------------------------------
+
+// returns a shared pointer to the texture with the given name or to the fallback texture
+const ResourceManager::TexturePtr& ResourceManager::getTexture(const std::string& name) const {
+    StringToTexture::iterator location;
+    IvTexture*                texture;
+    
+    location = textures.find(name);
+    if (location != textures.end()) {
+        texture_cache.remove(name);
+        texture_cache.push_back(name);
+        
+        return location->second;
+    }
+    
+    texture = loadTexture("../../Textures/" + name);
+    if (!texture)
+        return FALLBACK_TEXTURE;
+    
+    textures.insert(std::make_pair(name, TexturePtr(texture, CUSTOM_DELETER)));
+    texture_cache.push_back(name);
+    
+    if (textures.size() > SIZE_OF_CACHE) {
+        textures.erase(texture_cache.front());
+        texture_cache.pop_front();
+    }
+    
+    return textures[name];
+}
+
+// returns a shared pointer to the model with the given name or nullptr
+const ResourceManager::ConstAiScenePtr& ResourceManager::getModel(const std::string& name) const {
+    StringToAiScene::iterator   location;
+    const aiScene*              scene;
+    Assimp::Importer            importer;
+    
+    location = models.find(name);
+    if (location != models.end()) {
+        model_cache.remove(name);
+        model_cache.push_back(name);
+        
+        return location->second;
+    }
+    
+    importer.ReadFile("../../Models/" + name, aiProcess_Triangulate | aiProcess_FlipUVs);
+    scene = importer.GetOrphanedScene();
```

Comment:
Please leave a comment here on why GetOrphanedScene is used so it's more obvious that you are doing your own custom deletion.